src/ast/irbuilderbpf.cpp

```diff
@@ -2434,6 +2431,14 @@ llvm::Value *IRBuilderBPF::CreateDatastructElemLoad(
   return CreateIntCast(expr, getInt64Ty(), false);
 }
 
+LoadInst *IRBuilderBPF::CreateGlobalVarLoad(std::string_view var_name)
+{
+  auto *ld = CreateLoad(getInt64Ty(), module_.getGlobalVariable(var_name));
+  // if we want to modify these in user space the loads need to be volatile
+  ld->setVolatile(true);
```

Comment:
Yes, the volatile is key or else the bpf programs don't see the updates from userspace (this is due to the compiler doing some tricky stuff)

Reply:
`global->setExternallyInitialized(true);` might be a better representation for what we want here. Since it's in read-only memory and can't change, it isn't really a volatile variable

Reply:
Ah cool. I didn't know about `setExternallyInitialized`. It works, so I'll use that instead. Thanks!